css/css-display/focus/display-contents-focus.html

```diff
@@ -0,0 +1,50 @@
+<!doctype html>
+<html>
+<head>
+  <title>Testing focus for display: contents</title>
+  <link rel="help" href="https://drafts.csswg.org/css-display-4/#box-generation">
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+</head>
+<body>
+
+  <!-- Dec 2023 notes for "display: contents" testing:
+  - Per CSS spec, setting "display: contents" must not alter an element's semantics (https://www.w3.org/TR/css-display-3/#valdef-display-contents):
+  "As only the box tree is affected, any semantics based on the document tree, such as selector-matching, event handling, and
+  property inheritance, are not affected."
+
+  -->
+
+  <h1>Testing focusability of display: contents</h1>
+
+  <button style="display: contents;" class="ex-focusable" data-testname="button with display: contents is focusable">x</button>
+  <div role="button" tabindex="0" style="display: contents;" class="ex-focusable" data-testname="div with role button, tabindex=0 and display: contents is focusable"></div>
+  <div role="button" tabindex="-1" style="display: contents;" class="ex-focusable" data-testname="div with role button, tabindex=-1 and display: contents is focusable"></div>
+
+  <a href="#" style="display: contents;" class="ex-focusable" data-testname="link with display: contents is focusable">x</a>
+  <span role="link" tabindex="0" style="display: contents;" class="ex-focusable" data-testname="span with role link, tabindex=0 and display: contents is focusable"></span>
+
+<script>
+  verifyElementsAreKeyboardFocusable();
+
+  function verifyElementsAreKeyboardFocusable() {
+    const els = document.querySelectorAll(".ex-focusable");
+    if (!els.length) {
+      throw `Selector passed in verifyElementsAreKeyboardFocusable should match at least one element.`;
+    }
```

Comment:
I would leave this out. It should never be hit and makes it take a bit longer to read what is going on.

Reply:
I'd prefer he keep it. The problem we've seen (on multiple occasions) is that PR authors have misspelled one of the selectors and not noticed a the silent failure of a missing test result in a larger list of subtests... It's sometimes difficult for reviewers to spot minor selector errors, too.

Reply:
Maybe use `assert_unreached(msg)` instead of `throw`

Reply:
This is outside a test so asserts wouldn't work.

@cookiecrook this `querySelector()` is the static part of this test, no? Seems like it would not be changed anymore. It seems more likely someone would typo "ex-focusable" when adding a new element above, which we can't really guard against. Or maybe we could, but we aren't now.

Reply:
I may be misunderstanding your point, but I'm coming around to leaving it out... This is only a help for authors while writing the test, and a classname typo would still result in a silent failure unless _all_ of them had the same typo, so I can see your point about limited utility.

Reply:
- https://github.com/web-platform-tests/interop-accessibility/issues/102

Reply:
@annevk Also linking this PR as a potential avenue for improved error catching: https://github.com/web-platform-tests/wpt/pull/42769.

As there are other explorations, I'll leave the `display: contents` test as is but thanks for raising this issue!

Reply:
As long as that happens I guess that's okay.